scarb/src/compiler/compilers/executable.rs

```diff
@@ -0,0 +1,49 @@
+use crate::compiler::helpers::write_json;
+use crate::compiler::helpers::{build_compiler_config, collect_main_crate_ids};
+use crate::compiler::{CairoCompilationUnit, CompilationUnitAttributes, Compiler};
+use crate::core::{TargetKind, Workspace};
+use anyhow::{Context, Result};
+use cairo_lang_compiler::db::RootDatabase;
+use cairo_lang_executable::executable::Executable;
+
+pub struct ExecutableCompiler;
+
+impl Compiler for ExecutableCompiler {
+    fn target_kind(&self) -> TargetKind {
+        TargetKind::EXECUTABLE.clone()
+    }
+
+    fn compile(
+        &self,
+        unit: CairoCompilationUnit,
+        db: &mut RootDatabase,
+        ws: &Workspace<'_>,
+    ) -> Result<()> {
+        let target_dir = unit.target_dir(ws);
+        let main_crate_ids = collect_main_crate_ids(&unit, db);
+        let compiler_config = build_compiler_config(db, &unit, &main_crate_ids, ws);
+        let executable = Executable::new(
+            cairo_lang_executable::compile::compile_executable_in_prepared_db(
```

Comment:
Please add `trace_span` like [this](https://github.com/software-mansion/scarb/blob/1f85aa1811e8d849cc7c210e2ca510fac903000f/scarb/src/compiler/compilers/lib.rs#L66) to this call. This can be helpful for debugging Scarb execution.